perf/mhash.cc

```diff
@@ -0,0 +1,81 @@
+#include <algorithm>
+#include <array>
+#include <cstdlib>
+#include <numeric>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#define mh_name _u64
+#define mh_key_t uint64_t
+#define mh_node_t uint64_t
+#define mh_arg_t void *
+#define mh_hash(a, arg) (*(a))
+#define mh_hash_key(a, arg) (a)
+#define mh_cmp(a, b, arg) (*(a) != *(b))
+#define mh_cmp_key(a, b, arg) ((a) != *(b))
+#define MH_SOURCE
+#include <salad/mhash.h>
+
+class MHashU64Fixture : public benchmark::Fixture {
+public:
+	MHashU64Fixture() : h{nullptr}, filler_keys(filler_keys_count)
+	{
+		std::srand(0);
+		generate_rand_keys();
+
+		std::iota(filler_keys.begin(), filler_keys.end(), 0);
+	}
+
+	void
+	SetUp(const benchmark::State &state) override
+	{
+		h = ::mh_u64_new();
+		for (std::int64_t i = 0; i < state.range(0); ++i)
+			::mh_u64_put(h, &filler_keys[i], nullptr, nullptr);
+	}
+
+	void
+	TearDown(const benchmark::State &state) override
+	{
+		::mh_u64_delete(h);
+	}
+
+protected:
+	mh_u64_t *h;
+
+	static constexpr std::size_t rand_keys_count = 1 << 10;
+	std::array<std::uint64_t, rand_keys_count> rand_keys;
+
+	static constexpr std::size_t filler_keys_count = 1 << 30;
+	std::vector<std::uint64_t> filler_keys;
+
+	void
+	generate_rand_keys()
+	{
+		std::generate(rand_keys.begin(), rand_keys.end(), []() {
+			return filler_keys_count +
```

Comment:
Yes. What seems weird to you? This is an abstraction over the generation of random keys. For now it's `rand` from **libc**, in the future it could be the [C++11 random library](https://en.cppreference.com/w/cpp/numeric/random).

Reply:
Turned out the confusion was caused by the fact I actually test `find` over _absent_ keys. Supplied a comment and changed the naming correspondigly.